src/BikramSambat.test.ts

```diff
@@ -54,4 +54,53 @@ describe('BikramSambat Class', () => {
     const bikramSambat = new BikramSambat('2075-01-32')
     expect(bikramSambat.getDay()).toBe(NaN)
   })
+
+  it('should return previous year', () => {
+    const bikramSambat = new BikramSambat('2075-01-01')
+    expect(bikramSambat.getPreviousYear()).toBe(2074)
+  })
+
+  it('should return next year', () => {
+    const bikramSambat = new BikramSambat('2075-01-01')
+    expect(bikramSambat.getNextYear()).toBe(2076)
+  })
+
+  it('should return NaN for Invalid Date', () => {
+    const bikramSambat = new BikramSambat('2075-01-32')
+    expect(bikramSambat.getPreviousYear()).toBe(NaN)
+    expect(bikramSambat.getNextYear()).toBe(NaN)
```

Comment:
For Invalid date, isn't "Invalid date" string error response more appropriate and better for the UX? The NaN doesn't fit well for such case.

Reply:
Yes, you're right, if date is invalid then it will return "Invalid Date" but if you try to call method on that object then for instance for .getPreviousYear() which should return previous year as number, it returns NaN.
`"Invalid Date".getPreviousYear() = NaN`

Reply:
Okay, this looks good.